soh/src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -2170,7 +2170,7 @@ void func_80834298(Player* this, PlayState* play) {
 
 // Determine projectile type for bow or slingshot
 s32 func_80834380(PlayState* play, Player* this, s32* itemPtr, s32* typePtr) {
-    if (LINK_IS_ADULT) {
+    if ((CVar_GetS32("gSlingshotBowAmmoFix", 0) && this->heldItemActionParam != PLAYER_AP_SLINGSHOT) || !CVar_GetS32("gSlingshotBowAmmoFix", 0) && LINK_IS_ADULT) {
```

Comment:
Let’s divide this expression into separate vars that are easier to reason about and read

Reply:
```suggestion
    if ((CVar_GetS32("gSlingshotBowAmmoFix", 0) && 
    this->heldItemActionParam != PLAYER_AP_SLINGSHOT) || 
    !CVar_GetS32("gSlingshotBowAmmoFix", 0) && LINK_IS_ADULT) {
```
something like this?